### DISCUSSION TITLE
How to configure log format?

### DISCUSSION THREAD

**Discussion by oliviermichaelis** [renovatebot/renovate#29674] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, renovate 37.408.0

### Please tell us more about your question or problem

I would prefer renovate's log output to be JSON Lines so that the logs are parsed properly in my favorite log aggregation system.
Searching the documentation, I could only find the following:
https://github.com/renovatebot/renovate/blob/b07f8bc20fababb52e7b00fc5449bf057126b5f3/docs/usage/self-hosted-configuration.md?plain=1#L757

I tried setting `logFormat` to `json`, but I got the following validation error:
```
2024-06-14 12:16:05.073	WARN: Config validation errors found in /usr/src/app/configuration/config.js
2024-06-14 12:16:05.073	"errors": [
2024-06-14 12:16:05.073	  {
2024-06-14 12:16:05.073	    "topic": "Configuration Error",
2024-06-14 12:16:05.073	    "message": "Invalid configuration option: logFormat"
2024-06-14 12:16:05.073	  }
2024-06-14 12:16:05.073	]
```

I couldn't find any other reference in the code. Is it possible to set the log format to `json` somehow?
Thank you for your help!

**Comment by oliviermichaelis** [comment, marked as answer]
I found the [this discussion](https://github.com/renovatebot/renovate/discussions/16641) which hints at setting the environment variable `LOG_FORMAT` to `json`. That works for me :)

**Reply by HonkingGoose** [reply]
We also have these docs:

- [Renovate docs, Troubleshooting Renovate](https://docs.renovatebot.com/troubleshooting/)
- [Renovate docs, Self-hosting, logging](https://docs.renovatebot.com/examples/self-hosting/#logging)
